src/libraries/System.Text.RegularExpressions/tests/RegexCharacterSetTests.cs

```diff
@@ -152,7 +174,111 @@ public void UnicodeCategoryInclusionsExpected(string set, UnicodeCategory[] cate
+        [InlineData(@"\p{IsBraillePatterns}", new[] { 0x2800, 0x28FF })]
+        [InlineData(@"\p{IsSupplementalArrows-B}", new[] { 0x2900, 0x297F })]
+        [InlineData(@"\p{IsMiscellaneousMathematicalSymbols-B}", new[] { 0x2980, 0x29FF })]
+        [InlineData(@"\p{IsSupplementalMathematicalOperators}", new[] { 0x2A00, 0x2AFF })]
+        [InlineData(@"\p{IsMiscellaneousSymbolsandArrows}", new[] { 0x2B00, 0x2BFF })]
+        [InlineData(@"\p{IsCJKRadicalsSupplement}", new[] { 0x2E80, 0x2EFF })]
+        [InlineData(@"\p{IsKangxiRadicals}", new[] { 0x2F00, 0x2FDF })]
+        [InlineData(@"\p{IsIdeographicDescriptionCharacters}", new[] { 0x2FF0, 0x2FFF })]
+        [InlineData(@"\p{IsCJKSymbolsandPunctuation}", new[] { 0x3000, 0x303F })]
+        [InlineData(@"\p{IsHiragana}", new[] { 0x3040, 0x309F })]
+        [InlineData(@"\p{IsKatakana}", new[] { 0x30A0, 0x30FF })]
+        [InlineData(@"\p{IsBopomofo}", new[] { 0x3100, 0x312F })]
+        [InlineData(@"\p{IsHangulCompatibilityJamo}", new[] { 0x3130, 0x318F })]
+        [InlineData(@"\p{IsKanbun}", new[] { 0x3190, 0x319F })]
+        [InlineData(@"\p{IsBopomofoExtended}", new[] { 0x31A0, 0x31BF })]
+        [InlineData(@"\p{IsKatakanaPhoneticExtensions}", new[] { 0x31F0, 0x31FF })]
+        [InlineData(@"\p{IsEnclosedCJKLettersandMonths}", new[] { 0x3200, 0x32FF })]
+        [InlineData(@"\p{IsCJKCompatibility}", new[] { 0x3300, 0x33FF })]
+        [InlineData(@"\p{IsCJKUnifiedIdeographsExtensionA}", new[] { 0x3400, 0x4DBF })]
+        [InlineData(@"\p{IsYijingHexagramSymbols}", new[] { 0x4DC0, 0x4DFF })]
+        [InlineData(@"\p{IsCJKUnifiedIdeographs}", new[] { 0x4E00, 0x9FFF })]
+        [InlineData(@"\p{IsYiSyllables}", new[] { 0xA000, 0xA48F })]
+        [InlineData(@"\p{IsYiRadicals}", new[] { 0xA490, 0xA4CF })]
+        [InlineData(@"\p{IsHangulSyllables}", new[] { 0xAC00, 0xD7AF })]
+        [InlineData(@"\p{IsHighSurrogates}", new[] { 0xD800, 0xDB7F })]
+        [InlineData(@"\p{IsHighPrivateUseSurrogates}", new[] { 0xDB80, 0xDBFF })]
+        [InlineData(@"\p{IsLowSurrogates}", new[] { 0xDC00, 0xDFFF })]
+        [InlineData(@"\p{IsPrivateUse}", new[] { 0xE000, 0xF8FF })]
+        [InlineData(@"\p{IsCJKCompatibilityIdeographs}", new[] { 0xF900, 0xFAFF })]
+        [InlineData(@"\p{IsAlphabeticPresentationForms}", new[] { 0xFB00, 0xFB4F })]
+        [InlineData(@"\p{IsArabicPresentationForms-A}", new[] { 0xFB50, 0xFDFF })]
+        [InlineData(@"\p{IsVariationSelectors}", new[] { 0xFE00, 0xFE0F })]
+        [InlineData(@"\p{IsCombiningHalfMarks}", new[] { 0xFE20, 0xFE2F })]
+        [InlineData(@"\p{IsCJKCompatibilityForms}", new[] { 0xFE30, 0xFE4F })]
+        [InlineData(@"\p{IsSmallFormVariants}", new[] { 0xFE50, 0xFE6F })]
+        [InlineData(@"\p{IsArabicPresentationForms-B}", new[] { 0xFE70, 0xFEFF })]
+        [InlineData(@"\p{IsHalfwidthandFullwidthForms}", new[] { 0xFF00, 0xFFEF })]
+        [InlineData(@"\p{IsSpecials}", new[] { 0xFFF0, 0xFFFF })]
```

Comment:
This test tests against everything up to char.MaxValue of course. How should I think about regex's behavior in the face of surrogates? My belief was that our strings and all the API around them do not have an understanding of encoding a code point (hopefully I'm using the correct terminology) in more than 2 bytes. Yet we support testing for `IsHighSurrogates` which implies that in some cases developers may manually store such encoded code points in a string, and presumably deal with it with some care. Do we expect that to work? Or is `IsHighSurrogates` here just to cover the whole list?

Reply:
cc @GrabYourPitchforks

Reply:
As far as I'm aware, we have no special handling in S.T.RegularExpressions.dll for surrogate pairs, with the engine only examining one System.Char at a time.  This is simply one of the named character classes recognized as a range of values:
https://github.com/dotnet/runtime/blob/6a6ddf3ae356d304630b69032d98de68e2078a90/src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCharClass.cs#L195
Rather than taking it from the source, I took the list of supported named regions from the docs:
https://docs.microsoft.com/en-us/dotnet/standard/base-types/character-classes-in-regular-expressions#supported-named-blocks

Reply:
OK, as I thought - thanks. I suppose in theory one could write regexes to deal with these, but they might look ugly.